src/main/java/eu/webeid/example/security/WebEidAuthentication.java

```diff
@@ -52,12 +54,15 @@ private WebEidAuthentication(String principalName, String idCode, List<GrantedAu
     }
 
     private static String getPrincipalNameFromCertificate(X509Certificate userCertificate) throws CertificateEncodingException {
-        try {
-            return Objects.requireNonNull(CertificateData.getSubjectGivenName(userCertificate)) + ' ' +
-                    Objects.requireNonNull(CertificateData.getSubjectSurname(userCertificate));
-        } catch (CertificateEncodingException e) {
+        final Optional<String> givenName = CertificateData.getSubjectGivenName(userCertificate);
+        final Optional<String> surname = CertificateData.getSubjectSurname(userCertificate);
+
+        if (givenName.isPresent() && surname.isPresent()) {
+            return givenName.get() + ' ' + surname.get();
+        } else {
             // Organization certificates do not have given name and surname fields.
-            return Objects.requireNonNull(CertificateData.getSubjectCN(userCertificate));
+            return CertificateData.getSubjectCN(userCertificate)
```

Comment:
Is there a reason, why the .getSubjectCN(...) would throw an exception? In the java-validation repository the .getSubjectCN(...) calls .getSubjectField(...) which supposedly throws CertificateEncodingException, but the .getSubjectField(...) calls .getField(...) from which the "throws ...Exception" has been removed.

See [the PR for validation repository](https://github.com/web-eid/web-eid-authtoken-validation-java/pull/60/files) CertificateData.java file lines 60 and 64.